pkg/controller/backup_status_updater.go

```diff
@@ -102,28 +115,202 @@ func (u *realBackupConditionUpdater) Update(backup *v1alpha1.Backup, condition *
 
 // updateBackupStatus updates existing Backup status
 // from the fields in BackupUpdateStatus.
-func updateBackupStatus(status *v1alpha1.BackupStatus, newStatus *BackupUpdateStatus) {
+func updateBackupStatus(status *v1alpha1.BackupStatus, newStatus *BackupUpdateStatus) bool {
+	isUpdate := false
 	if newStatus == nil {
-		return
+		return isUpdate
 	}
-	if newStatus.BackupPath != nil {
+	if newStatus.BackupPath != nil && status.BackupPath != *newStatus.BackupPath {
 		status.BackupPath = *newStatus.BackupPath
+		isUpdate = true
 	}
-	if newStatus.TimeStarted != nil {
+	if newStatus.TimeStarted != nil && status.TimeStarted != *newStatus.TimeStarted {
 		status.TimeStarted = *newStatus.TimeStarted
+		isUpdate = true
 	}
-	if newStatus.TimeCompleted != nil {
+	if newStatus.TimeCompleted != nil && status.TimeCompleted != *newStatus.TimeCompleted {
 		status.TimeCompleted = *newStatus.TimeCompleted
+		isUpdate = true
 	}
-	if newStatus.BackupSizeReadable != nil {
+	if newStatus.BackupSizeReadable != nil && status.BackupSizeReadable != *newStatus.BackupSizeReadable {
 		status.BackupSizeReadable = *newStatus.BackupSizeReadable
+		isUpdate = true
 	}
-	if newStatus.BackupSize != nil {
+	if newStatus.BackupSize != nil && status.BackupSize != *newStatus.BackupSize {
 		status.BackupSize = *newStatus.BackupSize
+		isUpdate = true
 	}
-	if newStatus.CommitTs != nil {
+	if newStatus.CommitTs != nil && status.CommitTs != *newStatus.CommitTs {
 		status.CommitTs = *newStatus.CommitTs
+		isUpdate = true
+	}
+	// if newStatus.LogStopped != nil && status.LogStopped != *newStatus.LogStopped {
+	// 	status.LogStopped = *newStatus.LogStopped
+	// 	isUpdate = true
+	// }
+	if newStatus.LogCheckpointTs != nil && status.LogCheckpointTs != *newStatus.LogCheckpointTs {
+		status.LogCheckpointTs = *newStatus.LogCheckpointTs
+		isUpdate = true
+	}
+	if newStatus.LogTruncateUntil != nil && status.LogTruncateUntil != *newStatus.LogTruncateUntil {
+		status.LogTruncateUntil = *newStatus.LogTruncateUntil
+		isUpdate = true
+	}
+	return isUpdate
+}
+
+// updateLogBackupSubcommandStatus upserts log backup subcommand.
+// start command will update log backup status together.
+// truncate/stop command complete will update log backup status together.
+func updateLogBackupSubcommandStatus(backup *v1alpha1.Backup, condition *v1alpha1.BackupCondition, newStatus *BackupUpdateStatus) bool {
```

Comment:
don't write a too long function, keep function short

Reply:
ok, i will refine code, this push is to run e2e test to make sure all br test can pass.